frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright © 2015 Juan José Aguililla. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package sabina.benchmark;
+
+import static com.mongodb.client.model.Filters.eq;
+import static java.lang.Integer.parseInt;
+import static sabina.benchmark.Application.DB_ROWS;
+
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+
+final class MongoDbRepository implements Repository {
+    private static final String [] FORTUNES = {
+        "fortune: No such file or directory",
+        "A computer scientist is someone who fixes things that aren't broken.",
+        "After enough decimal places, nobody gives a damn.",
+        "A bad random number generator: 1, 1, 1, 1, 1, 4.33e+67, 1, 1, 1",
+        "A computer program does what you tell it to do, not what you want it to do.",
+        "Emacs is a nice operating system, but I prefer UNIX. — Tom Christaensen",
+        "Any program that runs right is obsolete.",
+        "A list is only as strong as its weakest link. — Donald Knuth",
+        "Feature: A bug with seniority.",
+        "Computers make very fast, very accurate mistakes.",
+        "<script>alert(\"This should not be displayed in a browser alert box.\");</script>",
+        "フレームワークのベンチマーク"
+    };
+
+    private MongoCollection<Document> worldCollection;
+    private MongoCollection<Document> fortuneCollection;
+
+    MongoDbRepository (Properties settings) {
+        final int PORT = parseInt (settings.getProperty ("mongodb.port"));
+        final String HOST = settings.getProperty ("mongodb.host");
+        final String DATABASE = settings.getProperty ("mongodb.database");
+        final String WORLD = settings.getProperty ("mongodb.world.collection");
+        final String FORTUNE = settings.getProperty ("mongodb.fortune.collection");
+
+        MongoClient mongoClient = new MongoClient (HOST, PORT);
+        MongoDatabase db = mongoClient.getDatabase (DATABASE);
+        worldCollection = db.getCollection (WORLD);
+        fortuneCollection = db.getCollection (FORTUNE);
+
+        loadData ();
```

Comment:
@jamming please remove this call and the associated `loadData` function + `FORTUNES` data. We already setup the databases for you.